File: src/main/java/de/take_weiland/mods/commons/internal/sync_olds/NetworkLinkedObjectContainer.java
package de.take_weiland.mods.commons.internal.sync_olds;

import de.take_weiland.mods.commons.net.MCDataInput;
import de.take_weiland.mods.commons.net.MCDataOutput;
import de.take_weiland.mods.commons.net.SimplePacket;
import de.take_weiland.mods.commons.sync.NetworkLinkedObjectType;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.inventory.Container;

/**
 * @author diesieben07
 */
public class NetworkLinkedObjectContainer implements NetworkLinkedObjectType<Container, Integer> {
    @Override
    public void write(Container object, MCDataOutput out) {
        out.writeByte(object.windowId);
    }

    @Override
    public Container read(MCDataInput in, EntityPlayer player) {
        int windowId = in.readInt();
        return getObject(windowId, player);
    }

    @Override
    public Integer getData(Container object) {
        return object.windowId;
    }

    @Override
    public Container getObject(Integer windowId, EntityPlayer player) {
        return player.openContainer.windowId == windowId ? player.openContainer : null;
    }

    @Override
    public void sendToTracking(Container object, SimplePacket packet) {
        packet.sendToViewing(object);
    }
}

File: src/main/java/de/take_weiland/mods/commons/internal/sync_olds/NetworkLinkedObjectContainer.kt
package de.take_weiland.mods.commons.internal.sync_olds

import de.take_weiland.mods.commons.net.MCDataInput
import de.take_weiland.mods.commons.net.MCDataOutput
import de.take_weiland.mods.commons.net.simple.SimplePacket
import de.take_weiland.mods.commons.sync.NetworkLinkedObjectType
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.inventory.Container

/**
 * @author diesieben07
 */
class NetworkLinkedObjectContainer : NetworkLinkedObjectType<Container, Int> {
    override fun write(`object`: Container, out: MCDataOutput) {
        out.writeByte(`object`.windowId)
    }

    override fun read(`in`: MCDataInput, player: EntityPlayer): Container? {
        val windowId = `in`.readInt()
        return getObject(windowId, player)
    }

    override fun getData(`object`: Container): Int? {
        return `object`.windowId
    }

    override fun getObject(windowId: Int?, player: EntityPlayer): Container? {
        return if (player.openContainer.windowId == windowId) player.openContainer else null
    }

    override fun sendToTracking(obj: Container, packet: SimplePacket) {
//        packet.sendToTracking(obj)
    }
}
